model/DataDifferencer.php

```diff
@@ -197,26 +199,29 @@ public function ChangedFields() {
 			if(!$this->fromRecord || $this->fromRecord->$field != $this->toRecord->$field) {
 				// Only show HTML diffs for fields which allow HTML values in the first place
 				$fieldObj = $this->toRecord->dbObject($field);
-				if($this->fromRecord) {
-					$fieldDiff = Diff::compareHTML(
-						$this->fromRecord->$field,
-						$this->toRecord->$field,
-						(!$fieldObj || $fieldObj->stat('escape_type') != 'xml')
-					);
-				} else {
-					if($fieldObj && $fieldObj->stat('escape_type') == 'xml') {
-						$fieldDiff = "<ins>" . $this->toRecord->$field . "</ins>";
+
+				if($fieldObj instanceof DBField) {
```

Comment:
`->toMap()` seems to return classes listed in the `$dependencies` properties as well, which could be regular classes.

Maybe dbObject() could check that before it returns the object found?